yt/config.py

```diff
@@ -16,8 +16,14 @@
 #-----------------------------------------------------------------------------
 
 import os
+import re
 import warnings
+from collections import defaultdict
 from yt.extern.six.moves import configparser
+from yt.utilities.exceptions import \
+    YTFieldNotFound
+
+SECTION_RE = re.compile('^config\.(?P<ftype>\w+)\.(?P<fname>\w+)$')
```

Comment:
@Xarthisius would you mind giving this regex a sanity check? I don't trust myself to :)

Reply:
FYI, the following strings
```
'config.gas.density'            # match
'config.gas_tracer.temperature' # match
'coonfig.doesnot.match'         # don't match
'config.does.not.match'         # don't match
' config.doesnot.match'         # don't match
'config.doesnot.match '         # don't match
```
But I'm no regexp guru so an extra eye won't hurt.